core/src/main/java/org/elasticsearch/search/suggest/SuggestUtils.java

```diff
@@ -208,78 +202,6 @@ public static StringDistance resolveDistance(String distanceVal) {
         public static final ParseField ACCURACY = new ParseField("accuracy");
    }
 
-    public static boolean parseDirectSpellcheckerSettings(XContentParser parser, String fieldName,
-                DirectSpellcheckerSettings suggestion, ParseFieldMatcher parseFieldMatcher) throws IOException {
-            if (parseFieldMatcher.match(fieldName, Fields.ACCURACY)) {
-                suggestion.accuracy(parser.floatValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.SUGGEST_MODE)) {
-                suggestion.suggestMode(SuggestUtils.resolveSuggestMode(parser.text()));
-            } else if (parseFieldMatcher.match(fieldName, Fields.SORT)) {
-                suggestion.sort(SortBy.resolve(parser.text()));
-            } else if (parseFieldMatcher.match(fieldName, Fields.STRING_DISTANCE)) {
-                suggestion.stringDistance(SuggestUtils.resolveDistance(parser.text()));
-            } else if (parseFieldMatcher.match(fieldName, Fields.MAX_EDITS)) {
-                suggestion.maxEdits(parser.intValue());
-                if (suggestion.maxEdits() < 1 || suggestion.maxEdits() > LevenshteinAutomata.MAXIMUM_SUPPORTED_DISTANCE) {
-                    throw new IllegalArgumentException("Illegal max_edits value " + suggestion.maxEdits());
-                }
-            } else if (parseFieldMatcher.match(fieldName, Fields.MAX_INSPECTIONS)) {
-                suggestion.maxInspections(parser.intValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.MAX_TERM_FREQ)) {
-                suggestion.maxTermFreq(parser.floatValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.PREFIX_LENGTH)) {
-                suggestion.prefixLength(parser.intValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.MIN_WORD_LENGTH)) {
-                suggestion.minWordLength(parser.intValue());
-            } else if (parseFieldMatcher.match(fieldName, Fields.MIN_DOC_FREQ)) {
-                suggestion.minDocFreq(parser.floatValue());
-            } else {
-                return false;
-            }
-            return true;
-    }
```

Comment:
As far as I can see, a lot of this parsing that was called from PhraseSuggestParser when parsing the `direct_generator` object has moved to DirectGeneratorBuilder. It is also still used on 2.x for TermSuggestParser, but that part of the code moved to TermSuggestBuilder which I see covers all the constants parsed here. So this is good to go I think.